httpmw/logging_mw.go

```diff
@@ -0,0 +1,70 @@
+package httpmw
+
+import (
+	"net/http"
+	"time"
+
+	"github.com/gorilla/mux"
+)
+
+type Logger interface {
+	Info(msg string, keysAndValues ...interface{})
+	Error(msg string, keysAndValues ...interface{})
+}
+
+func NewLogging(logger Logger) mux.MiddlewareFunc {
+	return func(next http.Handler) http.Handler {
+		return &loggingMiddleware{
+			next:   next,
+			now:    time.Now,
+			logger: logger,
+		}
+	}
+}
+
+var _ http.Handler = &loggingMiddleware{}
+
+type loggingMiddleware struct {
+	next   http.Handler
+	now    func() time.Time
+	logger Logger
+}
+
+type responseWriter struct {
+	http.ResponseWriter
+	statusCode int
+}
+
+func (rw *responseWriter) WriteHeader(code int) {
+	rw.statusCode = code
+	rw.ResponseWriter.WriteHeader(code)
+}
+
+func (rw *responseWriter) Write(b []byte) (int, error) {
+	if rw.statusCode == 0 {
+		rw.statusCode = http.StatusOK
+	}
+	return rw.ResponseWriter.Write(b)
+}
+
+func (l *loggingMiddleware) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	customWriter := &responseWriter{ResponseWriter: w}
```

Comment:
I think it probably is not hard to get the statusCode from w. Why bother to create responseWriter{}?

Reply:
The http.ResponseWriter interface does not expose the status code for us to easily access. 

https://stackoverflow.com/questions/53272536/how-do-i-get-response-statuscode-in-golang-middleware